include/fmt/os.h

```diff
@@ -379,56 +379,61 @@ struct ostream_params {
 #  endif
 };
 
+class file_buffer final : public detail::buffer<char> {
+  file file_;
+
+  FMT_API void grow(size_t) override;
+
+ public:
+  FMT_API file_buffer(cstring_view path, const detail::ostream_params& params);
+  FMT_API file_buffer(file_buffer&& other);
+  FMT_API ~file_buffer();
+
+  void flush() {
+    if (size() == 0) return;
+    file_.write(data(), size() * sizeof(data()[0]));
+    clear();
+  }
+
+  void close() {
+    flush();
+    file_.close();
+  }
+};
+
 FMT_END_DETAIL_NAMESPACE
 
 // Added {} below to work around default constructor error known to
 // occur in Xcode versions 7.2.1 and 8.2.1.
 constexpr detail::buffer_size buffer_size{};
 
 /** A fast output stream which is not thread-safe. */
-class FMT_API ostream final : private detail::buffer<char> {
+class FMT_API ostream final {
```

Comment:
`final` is no longer needed here because inheritance has been removed.

Reply:
```suggestion
class FMT_API ostream {
```